modules/custom/az_migration/migrations/migrate_plus.migration_group.az_migration.yml

```diff
@@ -0,0 +1,25 @@
+# The machine name of the group, by which it is referenced in individual migrations.
+id: az_migration
+
+# A human-friendly label for the group.
+label: AZ Quickstart Migrations
+
+# More information about the group.
+description: Common configuration for simple migrations.
+
+# Short description of the type of source, e.g. "Drupal 6" or "WordPress".
+source_type: Drupal 7 Site
+
+# Here we add any default configuration settings to be shared among all
+# migrations in the group.
+shared_configuration:
+ source:
+   key: migrate
+
+# We add dependencies just to make sure everything we need will be available
+dependencies:
+ enforced:
+   module:
+     - az_migration
+     - migrate_drupal
+     - migrate_tools
```

Comment:
After taking a closer look at this I'm thinking it might be best to put the module dependencies in the specific migration plugins:
- `az_node_person` and `az_person_categories` migrations should depend on the `az_person` module
- `az_cas_user` migration should depend on the `az_cas` module
- `az_media` migration should depend on the `az_media` module